### DISCUSSION TITLE
Compare with Selected is selecting an opened member over the one from right-click option

### DISCUSSION THREAD

**Discussion by m-tyler** [codefori/vscode-ibmi#754] [IBM i]
`Compare with Selected` is selecting an opened member over the one from right-click option.

So the demo shown in item #637, the selected compare member is open in a tab.   If I don't have the member I want to compare already open in a tab, I see what is in the images below.  If I don't have any members open, the `Compare with Selected` pop-up input fills out with the expected member string. 

Is this known or expected?  Its unexpected and I don't recall it happening before.  I just moved to a new laptop with a new version of VS Code.

**selecting member**
![image](https://user-images.githubusercontent.com/1353665/179268903-ddea5fd4-2685-4e4e-a1e1-c54e256a04dc.png)

**activating compare to input**
![image](https://user-images.githubusercontent.com/1353665/179317187-5996b80d-4f06-430d-aef3-d3add75c970e.png)

VS Code 1.69.1
Code for IBM I Version: v1.4.2

* QCCSID: 37
* Features enabled:
   * git: true
   * grep: true
   * tn5250: false
   * setccsid: true
   * GENCMDXML.PGM: true
   * QZDFMDB2.PGM: true
   * iconv: true
   * attr: true
* SQL enabled: true
* Source dates enabled: true

```json
[
  {
    "command": "system \"DLTOBJ OBJ(ILEDITOR/O_*) OBJTYPE(*FILE)\"",
    "code": 255,
    "stderr": "CPF2125: No objects deleted.",
    "cwd": "/home/PGMT"
  },
  {
    "command": "system \"CHKOBJ OBJ(QSYS/xCPTOIMPF) OBJTYPE(*DTAARA)\"",
    "code": 255,
    "stderr": "CPF9801: Object XCPTOIMPF in library QSYS not found.",
    "cwd": "/home/PGMT"
  },
  {
    "command": "system \"CHKOBJ OBJ(QSYS/xCPFRMIMPF) OBJTYPE(*DTAARA)\"",
    "code": 255,
    "stderr": "CPF9801: Object XCPFRMIMPF in library QSYS not found.",
    "cwd": "/home/PGMT"
  }
]
```
** Note: I disable the option to check for the old release setting for CPYTOIMPF/CPYFRMIMPF because we still use it that way but we built an override command to progress us out of using the 5.1 version of the commands.  That is why you see `QSYS/xCP` references.

**Comment by worksofliam** [comment]
@m-tyler It's a new change. If you run the 'Compare with Selected' command, the initial string in the box will be the active text editor path, and if there is no active editor then it will be the path for the initial file for comparison.

**Reply by m-tyler** [reply]
Ah, Okay now I know.

I use that feature all the time and I did not always have the member to compare as the active editor tab.

**Reply by worksofliam** [reply]
Another user suggested it'd be easier to navigate to the file first and then do the compare. Glad I explained it well enough 😄

**Reply by m-tyler** [reply]
I get the idea but it sure threw me.  I just quickly jumped to the library part of the path and change that, then press Enter to bring up the comparison.   I might have missed this change in the log if it was there.  

While I am on the topic of the compare with selected command, is it feasible to allow for variables a user supplies the value for and have the command default in those values for the library/source parts instead of the value from the member selected?  My reasoning is that Our environment has four source levels of libraries, developer, integration, quality and production.  With my project today I am most always changing the library value to the next environment, integration.  If I could supply a value for the library part in my connection profile it would definitely save me time and typos.

**Comment by Gregory1701** [comment]
When I right click and select a source member for compare, I get the toast message at the bottom that it was selected, but I don't get the input box at the top.  What am I doing wrong?
Version: 1.78.2 (user setup)
Commit: b3e4e68a0bc097f0ae7907b217c1119af9e03435
Date: 2023-05-10T14:39:26.248Z
Electron: 22.5.2
Chromium: 108.0.5359.215
Node.js: 16.17.1
V8: 10.8.168.25-electron.0
OS: Windows_NT x64 10.0.22000
Sandboxed: Yes

**Comment by worksofliam** [comment]
Yep! That's normal. It was added a long time ago.

**Reply by Gregory1701** [reply]
How do I get the compare with box to show then?
![image](https://github.com/halcyon-tech/vscode-ibmi/assets/132368707/9fdabd1d-e492-4b9d-b91b-7d15e4ffe506)

**Reply by worksofliam** [reply]
The input box will only appear if no file was first selected. E.g. if you 'Compare with Selected' without selecting another member first.

**Reply by Gregory1701** [reply]
I think I understand what it's supposed to do.  It's just not doing that for me:

https://github.com/halcyon-tech/vscode-ibmi/assets/132368707/af298f1a-3375-4af6-b748-4332db7e749c

**Reply by worksofliam** [reply]
@Gregory1701 Trying running 'Compare with Selected' from the command palette (F1)

**Reply by Gregory1701** [reply]
I got it to work.  Threw me off when you said that only comes up when you haven't previously selected a member for compare.  But when I select a member for compare, then select compare with selected in the command palette it comes up for me.  Thanks!
